cpp/ycm/CMakeLists.txt

```diff
@@ -103,16 +114,25 @@ if ( USE_CLANG_COMPLETER AND
     execute_process( COMMAND tar -xjf ${CLANG_FILENAME} )
   elseif( CLANG_FILENAME MATCHES ".+xz" )
     execute_process( COMMAND tar -xJf ${CLANG_FILENAME} )
+  elseif( CLANG_FILENAME MATCHES ".+exe" )
+    # NSIS tool is used to generate the Clang installer on Windows. Files
+    # from the installer can be extracted with 7-Zip.
+    execute_process( COMMAND 7z x ${CLANG_FILENAME} OUTPUT_QUIET )
```

Comment:
Does this make 7-zip a requirement for installation on Windows? If so do we need to document somewhere that 7-zip is a requirement for installing YCM?